### PR TITLE
Wizard recipe: OpenSSH-v8.1.0

### PR DESCRIPTION
**Pull request by StefanKarpinski**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: OpenSSH
* Version: v8.1.0

@staticfloat please review and merge.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
Don't merge. @keno is going to take a look at getting Windows working. musl platforms are also failing, so this is a bit of a mess.

File: O/OpenSSH/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("ssh-agent", :ssh_agent),
+    LibraryProduct("ssh", :ssh),
+    LibraryProduct("ssh-add", :ssh_add),
+    LibraryProduct("ssh-keygen", :ssh_keygen),
+    LibraryProduct("ssh-keyscan", :ssh_keyscan),
+    LibraryProduct("scp", :scp)
```
Comment (ViralBShah) [inline O/OpenSSH/build_tarballs.jl]: I think these are all `ExecutableProduct`s.

File: O/OpenSSH/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "OpenSSH"
+version = v"8.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/PowerShell/openssh-portable/archive/v8.1.0.0.tar.gz", "500276ab704c2a2a2b493b6fbb8b820bf954a54a77785eb81d34e03e1c086fac")
```
Comment (ViralBShah) [inline O/OpenSSH/build_tarballs.jl]: Maybe use `GithubSource`?

**Comment by DilumAluthge** [comment]
Bump. Any update on this?

**Comment by StefanKarpinski** [comment]
Nope. I tried a bit, then Viral took a crack at it. No luck for either of us.

**Comment by DilumAluthge** [comment]
Darn :(

I know that for Git, @giordano used the official Git for Windows binaries for Windows, and then we built from source for other platforms. I wonder if that would work here. Use official generic binaries for the platforms that we are having trouble with; otherwise build from source.

**Comment by DilumAluthge** [comment]
Would it be easier to build a subset of OpenSSH stuff? I know my use case (and the use case for the Documenter.jl people) is that we only need the functionality of `ssh-keygen`.

**Comment by Keno** [comment]
`ssh-keygen` is very easy to do in pure julia

**Comment by DilumAluthge** [comment]
> `ssh-keygen` is very easy to do in pure julia

You have any interest in making a package?

**Comment by DilumAluthge** [comment]
Will be useful for Documenter.jl, Franklin.jl, CompatHelper.jl, TagBot, etc. Basically anyone that needs to use SSH deploy keys.

**Comment by DilumAluthge** [comment]
Wait, how would this work? Won't you need a pure Julia implementation of RSA?

**Comment by DilumAluthge** [comment]
Isn't it better to use existing crypto libraries that have been audited, etc?

**Comment by ViralBShah** [comment]
Why not get this merged without Windows support for now?

**Comment by DilumAluthge** [comment]
> Why not get this merged without Windows support for now? And we definitely don't want to write our own `ssh-keygen`, however easy it may be!

Is Windows the only platform that doesn't work? I thought there were others.

For Windows, maybe we can just ship official binaries for PuTTY or something like that. That's what we did for Git_jll, and it works great.

**Comment by ViralBShah** [comment]
When I looked at this last windows had some real issues even though Microsoft seems to have a working Port. The others I am sure can be fixed relatively quickly.

**Comment by StefanKarpinski** [comment]
Windows is the only platform that can't be relied upon to have a working ssh binary...

**Comment by DilumAluthge** [comment]
@giordano How hard would it be to ship the official PuTTY binaries or something like that on Windows?

**Comment by DilumAluthge** [comment]
> Windows is the only platform that can't be relied upon to have a working ssh binary...

Yeah that's the irony, we need this on Windows most of all.

**Comment by DilumAluthge** [comment]
It's still nice to have on non-Windows platforms. Nice to not rely on the system OpenSSH not being broken.

**Comment by innerlee** [comment]
Can this be merged without the failed platforms?

**Comment by ViralBShah** [comment]
Definitely. We should probably give it one closer look to see if we can fix a few more platforms before merging.

**Comment by giordano** [comment]
`x86_64-linux-musl` should be easy to fix, not sure what's happening with macOS and FreeBSD, Windows is a mess.  Unfortunately, Windows is also the platform where this would be most needed.

**Comment by ViralBShah** [comment]
Upstream claims windows is supported... :-(

**Comment by giordano** [comment]
Upstream is literally Microsoft. It probably works with a very limited set of compilers (and given what's upstream...).

**Comment by StefanKarpinski** [comment]
If we're going to ship OpenSSH not supporting Windows, then perhaps we should use vanilla SSH instead of Microsoft's weirdo fork. The vanilla upstream doesn't have to wait for someone at Microsoft to rebase bug/security fixes. I originally had that working pretty easily but then switched to this upstream in an attempt to get Windows working... but of course, Windows doesn't work with the Microsoft fork of SSH 🤦‍♂️

**Comment by ViralBShah** [comment]
Their build instructions seem VS specific.

**Comment by ViralBShah** [comment]
Can we vendor the [binaries](https://github.com/PowerShell/Win32-OpenSSH/releases) they provide? 

There seem to be some [mingw instructions but those are quite old](https://github.com/PowerShell/Win32-OpenSSH/wiki/OpenSSH-64-bit-Build-and-Installation-Instructions).

Perhaps the fact that [mingw pacman](https://github.com/PowerShell/Win32-OpenSSH/releases) doesn't have an openssh recipe is telling. They have libssh and putty.

**Comment by giordano** [comment]
> Can we vendor the binaries they provide?

Not ideal, but that's maybe the best thing to do to get this working.  This is exactly what we have to do with Git

**Comment by StefanKarpinski** [comment]
Compiling vanilla OpenSSH on non-Windows and shipping Microsoft's binaries on Windows seems like a decent compromise. Not sure what the best way to do that is. Should I close this and let someone else take over?

**Comment by giordano** [comment]
Good to merge?

**Comment by ViralBShah** [comment]
Wow you got the windows binaries too! Merge it!

**Comment by giordano** [comment]
[`OpenSSH_jll`](https://github.com/JuliaBinaryWrappers/OpenSSH_jll.jl), [pull request](https://github.com/JuliaRegistries/General/pull/15713) to the General registry

**Comment by ViralBShah** [comment]
I merged it.

**Comment by StefanKarpinski** [comment]
Brilliant!

**Comment by aviks** [comment]
it is amazing that this now works, but for the record, Windows 10 now has a working openssh binary installed.

**Comment by ViralBShah** [comment]
Does windows 10 skip with openssh by default on all computers?

**Pull request merged**